packages/lib/shared/components/inputs/NumberInput.tsx

```diff
@@ -57,32 +61,56 @@ export function NumberInput({
         <Controller
           control={control}
           name={name}
-          render={({ field, fieldState }) => (
-            <>
-              <ChakraNumberInput
-                {...field}
-                isDisabled={isDisabled}
-                isInvalid={isInvalid || !!fieldState.error}
-                keepWithinRange={true}
-                onChange={field.onChange}
-                value={field.value}
-              >
-                <NumberInputField max={99} min={1} placeholder={placeholder} />
-              </ChakraNumberInput>
-              {isPercentage && (
-                <Text
-                  color="font.secondary"
-                  opacity={isDisabled ? 0.3 : 1}
-                  position="absolute"
-                  right="3"
-                  top="2.5"
-                  zIndex={1}
+          render={({ field, fieldState }) => {
+            const isSuggestionApplied = Number(field.value) === suggestedValue
+            return (
+              <>
+                <ChakraNumberInput
+                  {...field}
+                  isDisabled={isDisabled}
+                  isInvalid={isInvalid || !!fieldState.error}
+                  keepWithinRange={true}
+                  onChange={field.onChange}
+                  value={field.value}
                 >
-                  %
-                </Text>
-              )}
-            </>
-          )}
+                  <NumberInputField max={99} min={1} placeholder={placeholder} />
+                </ChakraNumberInput>
+                {isPercentage && (
+                  <Text
+                    color="font.secondary"
+                    opacity={isDisabled ? 0.3 : 1}
+                    position="absolute"
+                    right="3"
+                    top="2.5"
+                    zIndex={1}
+                  >
+                    %
+                  </Text>
+                )}
+
+                {suggestedValue && (
+                  <HStack justify="space-between" mt="xs" w="full">
+                    <HStack spacing="xs">
+                      <Text color="font.secondary" fontSize="sm">
+                        Suggested:
+                      </Text>
+                      <Text
+                        color={isSuggestionApplied ? 'font.secondary' : 'font.link'}
+                        cursor={isSuggestionApplied ? 'default' : 'pointer'}
+                        fontSize="sm"
+                        onClick={isSuggestionApplied ? undefined : onClickSuggestion}
+                        textDecoration={isSuggestionApplied ? 'none' : 'underline dotted 1px'}
+                        textUnderlineOffset="3px"
+                      >
+                        {suggestedValue}
+                      </Text>
+                    </HStack>
+                    {attribution && attribution}
```

Comment:
what is the attribution here? does it only make sense with the suggested value?

Reply:
attribution is for coingecko for price input

<img width="631" height="245" alt="image" src="https://github.com/user-attachments/assets/bc083c8f-f62d-456c-9ec5-0d93319dca06" />